File: c++/002.cc
#include <iostream>
#include <vector>
#include <numeric>

using namespace std;

int main()
{
  vector<int> fib;
  
  int a = 1, b = 1, c = 0;
  
  while (c <= 4000000)
  {
    c = a + b;
    if (c%2==0) fib.push_back(c);
    a = b;
    b = c;
  }
  
  int sum = accumulate(fib.begin(), fib.end(), 0);
  
  cout << sum << endl;
  
}

File: matlab/002.m
f = zeros(40,1);
f(1) = 1;
f(2) = 2;
for k = 3:40
    f(k) = f(k-1) + f(k-2);
end
sum(f(mod(f,2) == 0 & f < 4e6))